compiler/astalgo.nim

```diff
@@ -815,16 +815,21 @@ type
     name*: PIdent
 
 proc nextIdentIter*(ti: var TIdentIter, tab: TStrTable): PSym =
+  # hot spots
   var h = ti.h and high(tab.data)
   var start = h
-  result = tab.data[h]
-  while result != nil:
-    if result.name.id == ti.name.id: break
+  var p = addr tab.data[h]
```

Comment:
No. Instead use a `.cursor`.

Reply:
I used https://github.com/nim-lang/Nim/pull/20001/commits/e686303cd3e82c71334791b446345925e065c03e however it cannot pass CI

https://github.com/nim-lang/Nim/runs/7277971587?check_suite_focus=true

```
 /home/runner/work/Nim/Nim/compiler/astalgo.nim(821, 11) Error: invalid pragma: cursor
```

Reply:
Thank you!